### DISCUSSION TITLE
Welcome to Herbie (HRRR-B) Discussions!

### DISCUSSION THREAD

**Discussion by blaylockbk** [blaylockbk/Herbie#3] [General]
## 👋 Welcome!
I often receive emails asking about the HRRR archive and HRRR data. Instead of just helping that one person out, I'll try to use this Discussions page as a place to share my responses so others can benefit. This is also a place for you to talk about accessing and using HRRR data. 

We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

To get started, comment below with an introduction of yourself and tell us about what you do with HRRR data.

**Comment by alpinemike** [comment]
Hi Brian. You helped me out a while back with a HRRR sounding script. Definitely appreciated that. I've been going through a lot of your scripts and learning the tools of the trade. Noticed you created this HRRR-B module and I believe it would be very useful for me in my current project. Unfortunately I seem to be running into an issue and one that I've heard of before. 

I created a brand new environment in both Python 3.7 and 3.8 and downloaded HRRR-B via pip install off of github as it says in the documentation to do so. 

Here's what happens when I try: from `hrrrb.archive import download_hrrr`

```
/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/hrrrb/__init__.py:13: UserWarning: HRRR-B's xarray accessors could not be imported.
  warnings.warn("HRRR-B's xarray accessors could not be imported.")
Traceback (most recent call last):

  File "/Users/mikhailk/Documents/Python_Scripts/Sounding_Scripts/untitled0.py", line 9, in <module>
    from hrrrb.archive import download_hrrr, xhrrr

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/hrrrb/archive.py", line 69, in <module>
    import cfgrib

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/cfgrib/__init__.py", line 19, in <module>
    from .cfmessage import CfMessage

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/cfgrib/cfmessage.py", line 29, in <module>
    from . import messages

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/cfgrib/messages.py", line 28, in <module>
    import eccodes  # type: ignore

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/eccodes/__init__.py", line 13, in <module>
    from .eccodes import *  # noqa

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/eccodes/eccodes.py", line 12, in <module>
    from gribapi import (

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/gribapi/__init__.py", line 13, in <module>
    from .gribapi import *  # noqa

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/gribapi/gribapi.py", line 34, in <module>
    from . import errors

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/gribapi/errors.py", line 16, in <module>
    from .bindings import ENC, ffi, lib

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/gribapi/bindings.py", line 34, in <module>
    raise RuntimeError("Cannot find the ecCodes library")

RuntimeError: Cannot find the ecCodes library
```
Hopefully you know what the issue here is as I'm not sure where to go from here at this point. Thanks!

**Reply by alpinemike** [reply]
I seem to have figured out the eccodes issue actually. It appears as though when I download from github directly it does not install the correct version of eccodes as there seems to be some trouble with that right now. 


My next issue is quite odd as the code outputs exactly what is expected but then doesn't do what it should: 
```
💡 Info: Downloading [2] GRIB2 files

Download subset from [google]:
  Downloading GRIB line [  5]: variable=VIS, level=surface, forecast=12 hour fcst
✅ Success! Searched for [:VIS:surface] and got [1] GRIB fields and saved as /users/mikhailk/Desktop/hrrr/20201222/subset_20201222_hrrr.t01z.wrfsfcf12.grib2
🚛💨 Download Progress: [1/2 completed] >> Est. Time Remaining 0:00:01.505325  


🍦 Finished 🍦  Time spent on download: 0:00:01.505325
```
While it found 2 files it only downloaded one of them. Is my implementation incorrect? 
```
sDATE = datetime(2020, 12, 22, 1)
DATES = sDATE

# Download full GRIB2 files to local disk
download_hrrr(DATES, searchString=':VIS:surface', 
              fxx=[12,14],
              model='hrrr',
              field='sfc',
              save_dir="/users/mikhailk/Desktop",
              download_source_priority=None,
              dryrun=True, verbose=True)
```             
Thanks for the help!

**Reply by blaylockbk** [reply]
Hi!

That looks right from what I can tell. I haven't been at my computer in a week but will look into this when I can. 

I started writing a new API that might work better. If you recently installed the package from GitHub you should be able to use this example for a bulk download using the new api.

https://blaylockbk.github.io/HRRR_archive_download/_build/html/user_guide/tutorial.html#Bulk-Download

Thanks for bringing this to my attention. I'm still learning how to publish python packages and this will help me fix some things with it.

**Reply by alpinemike** [reply]
Hi Brian. 

I gave the Herbie Bulk Download function a go and unfortunately that didn't go very far. I immediately get a syntax error: 
```
runfile('/Users/mikhailk/Documents/Python_Scripts/Sounding_Scripts/test.py', wdir='/Users/mikhailk/Documents/Python_Scripts/Sounding_Scripts')
⚙ Created config file [/Users/mikhailk/.config/herbie/config.cfg] with default download directory set as [/Users/mikhailk/data] and default source priority as 
/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/herbie/__init__.py:13: UserWarning: heribe xarray accessors could not be imported.
  warnings.warn("heribe xarray accessors could not be imported.")
  File "<fstring>", line 1
    (found_grib=)
               ^
SyntaxError: invalid syntax
```

I then just tried to look for the file and it throws the same error with a little more: 
```
from herbie.tools import bulk_download
from herbie.tools import Herbie
from datetime import datetime

sDATE = datetime(2020, 12, 22, 1)
DATES = sDATE

H = Herbie(sDATE)
Traceback (most recent call last):

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 3441, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-4-5c8714b9e4a8>", line 1, in <module>
    from herbie.tools import bulk_download

  File "/opt/anaconda3/envs/Herbie/lib/python3.7/site-packages/herbie/tools.py", line 13, in <module>
    from herbie.archive import Herbie

  File "<fstring>", line 1
    (found_grib=)
               ^
SyntaxError: invalid syntax
```

Any advice on this would be great! Thanks.

**Reply by blaylockbk** [reply]
Oh, I think I know what this error is. I started using a Python 3.8 feature with printing f-strings. Maybe if you use python 3.8 this error will go away. I might have time to look into this more tomorrow.

See here: https://www.geeksforgeeks.org/new-operator-in-python3-8-f-string/

**Reply by alpinemike** [reply]
That finally did the trick! Thanks Brian. For whatever reason my Spyder runs very slow when I'm running Python anything greater than 3.7. But at least it does work.
